### ISSUE TITLE
dupeGuru PE OS X: scrolling is extremely slow in large dupe sets

### ISSUE THREAD

**Issue by deutrino** [arsenetar/dupeguru#259]
Scrolling, by dragging the scrollbar or clicking in the scrollbar region, is extremely slow with sets of dupes 6,000 or more on OS X. Click lag can be a second or two and every click is registered, so if you click 5 times you're waiting on the UI to catch up.  This behavior is most evident when scrolling more than just a couple pages away; nearby items (clicking once or twice on the scrollbar) or "seen" items seem to work faster, as if they were cached.

dupeGuru is not using a lot of CPU time during the lag.

This is OS X 10.7.  I may upgrade soon and hopefully remember to report whether this is the case on a newer version.

**Comment by stuckj** [comment]
I can repro this behavior as well. With 15000+ results on a network drive. I'm guessing it's network I/O related, but that's only a guess. I tried hiding the size and dimension columns thinking perhaps as you scroll it's getting these in real time from the filesystem, but that had no effect (although I suppose maybe you could still be getting them and just not displaying them). I can attest it is pretty annoying. :-P But, given that there's basically no other program that can do this, it's still worth using. :)

**Comment by stuckj** [comment]
Also, using a filter to reduce the number of results < 1000 doesn't help.

**Comment by hsoft** [comment]
Slow IOs might be a clue, yeah. @stuckj I have a two questions:

1. The results you're scrolling into, are they fresh off a scan, or did you load results that you've saved earlier?
2. If you scroll back to results you've already scrolled into, is the speed better (row data presented in the results table is cached, so the answer should normally be yes, but still, I'd like to know).

Thanks.

**Comment by arsenetar** [comment]
This is a rather old version, I am closing all issues related to 4.0.3 and older as there have been some fairly significant changes since.  Try the latest release at https://github.com/arsenetar/dupeguru/releases/latest, if you have a similar issue feel free to open a new issue.
